### ISSUE TITLE
cmd/cgo: can't pass kCFRunLoopDefaultMode as CFRunLoopMode

### ISSUE THREAD

**Issue by LiamHaworth** [golang/go#24531]
### What version of Go are you using (`go version`)?

`go version go1.10 darwin/amd64`

**and**

`go version devel +dc3a92edaa Mon Mar 26 19:42:08 2018 +0000 darwin/amd64`

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

```
GOARCH="amd64"
GOBIN="/Volumes/Storage/git/git/go_path/bin"
GOCACHE="/Users/liam/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Volumes/Storage/git/git/go_path"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.10/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.10/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/xx/mdt3785n0js3yrk67_ftskdr0000gn/T/go-build744503608=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

### What did you do?

Trying to compile CGO code on darwin that makes calls to C functions, it would compile correctly on go 1.9.2 but now fails to build on go 1.10

### What did you expect to see?

A successful build

### What did you see instead?

```
# command-line-arguments
./main.go:150: cannot use *_Cvar_kCFRunLoopDefaultMode (type _Ctype_CFStringRef) as type *_Ctype_struct___CFString in argument to func literal
./main.go:157: cannot use *_Cvar_kCFRunLoopDefaultMode (type _Ctype_CFStringRef) as type *_Ctype_struct___CFString in argument to func literal
```

### More information

I understand in go 1.10 there where changes made to how CGO is handled for darwin platforms, I did update some other parts of my code to comply with the new nil variable setting (`val =0` vs `val = nil`) but this error doesn't seem to make sense.

The function being called is [CFRunLoopRemoveSource](https://developer.apple.com/documentation/corefoundation/1542145-cfrunloopremovesource?language=objc) who's third parameter accepts a type of [CFRunLoopMode](https://developer.apple.com/documentation/corefoundation/cfrunloopmode?language=objc) which is just a typedef of CFStringRef.

For the third parameter I am passing [kCFRunLoopDefaultMode](https://developer.apple.com/documentation/corefoundation/kcfrunloopdefaultmode?language=objc) which should comply with the type matching as it has a type of CFRunLoopMode.

### Example Code

https://gist.github.com/LiamHaworth/d50ce9167fe0377cd20d6585d77a70b4

**Comment by LiamHaworth** [comment]
To add to this, when a dummy function is specified in Go code with the name signature it compiles without a complaint.

```go
func CFRunLoopRemoveSource(rl C.CFRunLoopRef, source C.CFRunLoopSourceRef, mode C.CFRunLoopMode)
```

**Comment by LiamHaworth** [comment]
The C signature is defined in the CoreFoundation library in this file https://opensource.apple.com/source/CF/CF-744/CFRunLoop.h on line 81

**Comment by andybons** [comment]
@ianlancetaylor

**Comment by ianlancetaylor** [comment]
CC @randall77

**Comment by ianlancetaylor** [comment]
@randall77 Would be nice to fix this for 1.11 if possible.  I hope that you have more state on this stuff than I do.

**Comment by randall77** [comment]
Dup of #24161
